go-getting-started/main.go

```diff
@@ -1,300 +1,31 @@
-	}
-}
-
-func (j jokesHandler) randomJokes(w http.ResponseWriter, r *http.Request) {
-	w.Header().Set("Content-Type", "application/json")
-
-	var jokes []models.Joke
-
-	pagination, err := j.parseSkipAndLimit(w, r)
-	if err != nil {
-		fmt.Println(err)
-	}
-
-	pipeline := bson.D{{Key: "$sample", Value: bson.D{{Key: "size", Value: pagination.Limit}}}}
-
-	cursor, err := j.collection.Aggregate(context.TODO(), mongo.Pipeline{pipeline})
-	if err != nil {
-		fmt.Println(err)
-	}
-
-	if err = cursor.All(context.TODO(), &jokes); err != nil {
-		panic(err)
-	}
-
-	if err = json.NewEncoder(w).Encode(jokes); err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-	}
-}
-
-func (j jokesHandler) funniest(w http.ResponseWriter, r *http.Request) {
-	w.Header().Set("Content-Type", "application/json")
-
-	var jokes []models.Joke
-
-	pagination, err := j.parseSkipAndLimit(w, r)
-	if err != nil {
-		fmt.Println(err)
-	}
-
-	findOptions := options.Find()
-	findOptions.SetSort(bson.D{{Key: "score", Value: -1}}).SetLimit(int64(pagination.Limit))
-
-	cursor, err := j.collection.Find(context.TODO(), bson.D{}, findOptions)
-	if err != nil {
-		w.WriteHeader(http.StatusInternalServerError)
-		return
-	}
-
-	if err = cursor.All(context.TODO(), &jokes); err != nil {
-		log.Fatal(err)
-	}
-
-	err = json.NewEncoder(w).Encode(jokes)
-	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-	}
-}
-
-func (j jokesHandler) search(w http.ResponseWriter, r *http.Request) {
-	w.Header().Set("Content-Type", "application/json")
-
-	search := r.URL.Query().Get("search")
-
-	var jokes []models.Joke
-
-	query := bson.M{
-		"title": bson.M{
-			"$regex": primitive.Regex{
-				Pattern: search,
-				Options: "i",
-			},
-		},
-	}
-
-	cursor, err := j.collection.Find(context.TODO(), query)
-	if err != nil {
-		w.WriteHeader(http.StatusInternalServerError)
-		return
-	}
-	for cursor.Next(context.TODO()) {
-		var joke models.Joke
-		err := cursor.Decode(&joke)
-		if err != nil {
-			panic(err)
-		}
-		jokes = append(jokes, joke)
-	}
-
-	err = json.NewEncoder(w).Encode(jokes)
-	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-	}
-}
-
-func (j jokesHandler) createJoke(w http.ResponseWriter, r *http.Request) {
-	if r.Method == http.MethodGet {
-		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
-		return
-	}
-
-	res, err := controllers.InsertData(j.collection, models.Joke{Body: "sdfsf", ID: "Sdfs", Score: 3, Title: "4324"})
-	if err != nil {
-		fmt.Println(err)
-	}
-
-	fmt.Println(res)
-}
+const port = ":8080"
 
 func main() {
 	content, _ := ioutil.ReadFile("reddit_jokes.json")
-	jokes := []models.Joke{}
+	jokes := []api.Joke{}
 	err := json.Unmarshal(content, &jokes)
 	if err != nil {
 		fmt.Println(err)
 	}
 
-	collection := config.ConnectDB(os.Getenv("MONGODB_URI"))
-	jh := jokesHandler{jokes, collection}
+	_ = storage.ConnectDB(os.Getenv("MONGODB_URI"))
```

Comment:
Why did you omit the returned argument? You should take this storage and embed it into the handler, in order to have access to its methods.